googlesource/googlesource.go

```diff
@@ -27,42 +29,69 @@ import (
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	googlebigquery "github.com/neha-Gupta1/conduit-connector-bigquery"
 	"google.golang.org/api/iterator"
-	"google.golang.org/api/option"
 )
 
 var (
 	newClient = bigquery.NewClient
```

Comment:
It is getting used as - https://github.com/neha-Gupta1/conduit-connector-
https://github.com/neha-Gupta1/conduit-connector-bigquery/blob/614c51973b097aa8bcc94c5c87a421f338210e59/googlesource/source_unit_test.go#L187-L194

We override it locally in the scope of a test case so it won't impact any other test case. Also, we don't run test cases in parallel so no chances of any such error.

Reply:
I'd say that this is still a global variable (and not a constant), which is a code smell.

> We override it locally in the scope of a test case so it won't impact any other test case.

What do you actually mean by this? `newClient` is still accessible to other tests (and a lot of production code). We do revert the change, but while the test is running, the changes are visible to everything else.

> Also, we don't run test cases in parallel so no chances of any such error.

That's true, I agree with that. However, I'd actually call that a limitation: it would be really great to be able to run tests in parallel, especially the slow ones.

Reply:
That is true. Is there any preferable way you want me to implement it? It is just used in a test case to mock and check in case of an unknown error.

Reply:
I'd suggest something like in the [first comment](https://github.com/neha-Gupta1/conduit-connector-bigquery/pull/12#discussion_r883808180). Basically, you create an interface for the BQ client and expose the needed functionality. The production impl. uses Google's BQ client, and in tests you can have a mock client. For example:
```go
type Client interface {
	ListTables() []string
}
``` 
In the production version, you'd use Google's client. In a test, you just mock it, and return whatever you want in that test.

Reply:
Implemented an interface https://github.com/neha-Gupta1/conduit-connector-bigquery/blob/691c1a9965979a0789a9f57f85f946b52ef139d0/googlesource/source.go#L115 which will take care of it.